packages/auth0_api_python/tests/test_api_client.py

```diff
@@ -388,3 +405,1137 @@ async def test_verify_access_token_fail_no_audience_config():
+            proof=dpop_proof,
+            http_method="GET",
+            http_url="https://api.example.com/resource"
+        )
+    assert "missing required claim: jti" in str(err.value).lower()
+
+@pytest.mark.asyncio
+async def test_verify_dpop_proof_fail_htm_mismatch():
+    """
+    Test that a DPoP proof with mismatched 'htm' claim fails verification.
+    """
+    access_token = "test_token"
+    dpop_proof = await generate_dpop_proof(
+        access_token=access_token,
+        http_method="POST",  # Generate proof for POST
+        http_url="https://api.example.com/resource",
+    )
+
+    api_client = ApiClient(
+        ApiClientOptions(domain="auth0.local", audience="my-audience")
+    )
+
+    with pytest.raises(InvalidDpopProofError) as err:
+        await api_client.verify_dpop_proof(
+            access_token=access_token,
+            proof=dpop_proof,
+            http_method="GET",  # But verify with GET
+            http_url="https://api.example.com/resource"
+        )
+
+    assert "htm mismatch" in str(err.value).lower()
+
+# --- HTU (HTTP URI) Validation Tests ---
+
+@pytest.mark.asyncio
+async def test_verify_dpop_proof_fail_htu_mismatch():
+    """
+    Test that a DPoP proof with mismatched 'htu' claim fails verification.
+    """
+    access_token = "test_token"
+    dpop_proof = await generate_dpop_proof(
+        access_token=access_token,
+        http_method="GET",
+        http_url="https://api.example.com/wrong-resource",  # Generate proof for wrong URL
+    )
+
+    api_client = ApiClient(
+        ApiClientOptions(domain="auth0.local", audience="my-audience")
+    )
+
+    with pytest.raises(InvalidDpopProofError) as err:
+        await api_client.verify_dpop_proof(
+            access_token=access_token,
+            proof=dpop_proof,
+            http_method="GET",
+            http_url="https://api.example.com/resource"  # But verify with correct URL
+        )
+
+    assert "htu mismatch" in str(err.value).lower()
+
+@pytest.mark.asyncio
+async def test_verify_dpop_proof_htu_url_normalization_case_sensitivity():
+    """
+    Test HTU URL normalization handles case sensitivity correctly.
+    """
+    access_token = "test_token"
+
+    # Test with different case in domain (should be normalized and pass)
+    dpop_proof = await generate_dpop_proof(
+        access_token=access_token,
+        http_method="GET",
+        http_url="https://API.EXAMPLE.COM/resource"  # Uppercase domain
+    )
+
+    api_client = ApiClient(
+        ApiClientOptions(domain="auth0.local", audience="my-audience")
+    )
+
+    # This should succeed due to URL normalization
+    result = await api_client.verify_dpop_proof(
+        access_token=access_token,
+        proof=dpop_proof,
+        http_method="GET",
+        http_url="https://api.example.com/resource"  # Lowercase domain
+    )
+    assert result is not None
+
+
+@pytest.mark.asyncio
+async def  test_verify_dpop_proof_htu_trailing_slash_mismatch():
```

Comment:
Please add another test case showing that when the query params and fragments are removed, the trailing slash is preserved.

E.g:
```
got = 'https://api.example.com/resource/?abc=def'
expected = 'https://api.example.com/resource/'
```